File: OPAL/br/src/main/scala/org/opalj/br/NestHost.scala
/* BSD 2-Clause License - see OPAL/LICENSE for details. */
package org.opalj
package br

/**
 * Definition of a Java 11 nest host.
 *
 * @author Dominik Helm
 */
case class NestHost(hostClassType: ObjectType) extends Attribute {

    final override def kindId: Int = NestHost.KindId

    override def similar(other: Attribute, config: SimilarityTestConfiguration): Boolean = {
        this == other
    }
}

object NestHost {

    final val KindId = 47

}

File: OPAL/bi/src/test/fixtures-java/projects/java11nests/NestHost.java
/* BSD 2-Clause License - see OPAL/LICENSE for details. */
package java11nests;

public class NestHost {

    private void nestHostMethod() {}

    class NestMember1 {
        private void nestMember1Method() {}
    }

    class NestMember2 {}

}

class NoNestMember {
    private void noNestMemberMethod() {}
}